### ISSUE TITLE
[Documentation] Notice on README about limitations?

### ISSUE THREAD

**Issue by saona-raimundo** [latex-lsp/tree-sitter-latex#80]
Hi! Thanks for the crate!

I wanted to ask if you would consider adding some words about the grammar being parsed, how it relates to (La)TeX and what are the limitations.
I know (La)Tex is super hard to parse, as the parser can be reconfigured during parser... 
But it would be nice to know your approach and limitations.

On a side note, I would really enjoy a "development blog" on what were the difficulties while developing the tree-sitter grammar for LaTeX.
I would like to give developing a parser for LaTeX-like syntax a try.

**Comment by pfoerster** [comment]
@saona-raimundo I have added a section in the README regarding some limitations.

> I know (La)Tex is super hard to parse, as the parser can be reconfigured during parser...

Yes, the only correct way to write a LaTeX parser would be to write a TeX engine that "compiles" the document from start to finish and outputs every step it does along the way. This is a very complicated task so this grammar and `texlab` uses a best-effort approach instead. We do not try to parse constructs like `catcode` and focus on the constructs specific to LaTeX (and not TeX).

This grammar originates from the language server so performance is also very important. Potential infinite loops or reparsing every document on every keystroke (which would be needed when going with the TeX engine approach unless you find a way to make the procesing incremental) are undesirable in an editing context.

> I would like to give developing a parser for LaTeX-like syntax a try.

The difficulty highly depends on the degree of correctness you want to achieve. This parser (and the one in `texlab`) only parse a subset of LaTeX, which is easy to parse. If you want to go the complete route, I recommend looking at [`tectonic`](https://github.com/tectonic-typesetting/tectonic). Maybe you can fork it and remove the generation of the output and output the macro expansions along the way.
